src/Frontend/test/specs/monitoring/license-enforcement.spec.ts

```diff
@@ -0,0 +1,116 @@
+import { expect } from "vitest";
+import { test, describe } from "../../drivers/vitest/driver";
+import * as precondition from "../../preconditions";
+import { expiredLicenseMessageWithValue } from "./questions/expiredLicenseMessageWithValue";
+import { viewYourLicenseButton } from "./questions/viewYourLicenseButton";
+import { extendYourLicenseButton } from "./questions/extendYourLicenseButton";
+import { getAlertNotifications } from "./questions/alertNotifications";
+
+describe("FEATURE: EXPIRING license detection", () => {
+  describe("RULE: The user should be alerted while using the {monitoring endpoint list} functionality about an EXPIRING license", () => {
+    test("EXAMPLE: Expiring trial", async ({ driver }) => {
+      //Arrange
+      await driver.setUp(precondition.serviceControlWithMonitoring);
+      await driver.setUp(precondition.hasExpiringLicense(precondition.LicenseType.Trial));
+
+      await driver.goTo("monitoring");
+
+      const notification = (await getAlertNotifications()).find(async (n) => {
+        n.textMatches(/your non\-production development license will expire soon\. to continue using the particular service platform you'll need to extend your license\./i);
+      });
+
+      expect(alert).not.toBeUndefined();
+      expect(notification?.hasLink({ caption: "Extend your license", address: "http://particular.net/extend-your-trial?p=servicepulse" })).toBeTruthy();
+      expect(notification?.hasLink({ caption: "View license details", address: "#/configuration" })).toBeTruthy();
+    });
+
+    [
+      { description: "Expiring upgrade protection", linceseType: precondition.LicenseType.UpgradeProtection },
+      { description: "Expiring platform subscription", linceseType: precondition.LicenseType.Subscription },
+    ].forEach(({ description, linceseType }) => {
+      test(`EXAMPLE: ${description}`, async ({ driver }) => {
+        //Arrange
+        await driver.setUp(precondition.serviceControlWithMonitoring);
+        await driver.setUp(precondition.hasExpiringLicense(linceseType));
+
+        await driver.goTo("monitoring");
+
+        const notification = (await getAlertNotifications()).find(async (n) => {
+          n.textMatches(/once upgrade protection expires, you'll no longer have access to support or new product versions/i);
+        });
+
+        expect(alert).not.toBeUndefined();
+        expect(notification?.hasLink({ caption: "View license details", address: "#/configuration" })).toBeTruthy();
+      });
+    });
+  });
+});
+
+describe("FEATURE: EXPIRED license detection", () => {
+  const serviceControlInstanceUrl = window.defaultConfig.service_control_url;
+
+  //As of the moment of writing this test, license check is performed during the first load of the application only. No continuous check is performed.
+  describe("RULE: Access to the {monitoring endpoint list} functionality should be blocked when a expired license is detected and a notification should be displayed", () => {
```

Comment:
```suggestion
  describe("RULE: Access to the monitoring endpoint list functionality should be blocked when a expired license is detected and a notification should be displayed", () => {
```
Are the { } to indicate that this is a component?